src/main/java/tick/tasks/Deadline.java

```diff
@@ -0,0 +1,43 @@
+package tick.tasks;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
+/**
+ * Represents a deadline task which has a deadline.
+ */
+public class Deadline extends Task {
+    private LocalDate by;
+
+    /**
+     * Constructs a deadline task with the given description and deadline.
+     *
+     * @param description Description of the deadline task.
+     * @param by Deadline of the task.
+     */
+    public Deadline(String description, LocalDate by) {
+        super(description);
+        this.by = by;
+    }
+
+    /**
+     * Converts the deadline task to a string format suitable for file storage.
+     * The format is "D | [status] | description | deadline".
+     *
+     * @return A string representation of the deadline task for file storage.
+     */
+    public String toFileString() {

```

Comment:
Should consider adding `@Override` annotation. Same goes for other overriding methods too :)